### ISSUE TITLE
Issue on different patch size

### ISSUE THREAD

**Issue by KSJhon** [HaoZhang1018/GTP-PNet#1]
Dear!
I am also working at pansharpening and trying to test your project with my own training set.
After a little modification, I successfully trained your TNET, but i have met a problem in PNET, especially in function "fusion_model" from line 223 to line 262.
    with tf.variable_scope('fusion_model'):
        MS_x2=tf.image.resize_images(images=img_MS, size=[2*self.image_size_MS, 2*self.image_size_MS],method=tf.image.ResizeMethod.BICUBIC)
        with tf.variable_scope('layer1_MS_x2'):
            weights=tf.get_variable("w1_MS_x2",[3,3,4,16],initializer=tf.truncated_normal_initializer(stddev=1e-3))
            bias=tf.get_variable("b1_MS_x2",[16],initializer=tf.constant_initializer(0.0))
            conv1_MS_x2=tf.nn.conv2d(MS_x2, weights, strides=[1,1,1,1], padding='SAME') + bias
            conv1_MS_x2 = lrelu(conv1_MS_x2) 
        MS_x4=tf.image.resize_images(images=conv1_MS_x2, size=[4*self.image_size_MS, 4*self.image_size_MS],method=tf.image.ResizeMethod.BICUBIC)
        with tf.variable_scope('layer2_MS_x4'):
            weights=tf.get_variable("w2_MS_x4",[3,3,16,4],initializer=tf.truncated_normal_initializer(stddev=1e-3))
            bias=tf.get_variable("b2_MS_x4",[4],initializer=tf.constant_initializer(0.0))
            conv2_MS_x4=tf.nn.conv2d(MS_x4, weights, strides=[1,1,1,1], padding='SAME') + bias
            conv2_MS_x4 = lrelu(conv2_MS_x4)       
######################################################### 
#########################################################       
        with tf.variable_scope('layer1_MS'):
            weights=tf.get_variable("w1_MS",[3,3,4,16],initializer=tf.truncated_normal_initializer(stddev=1e-3))
            bias=tf.get_variable("b1_MS",[16],initializer=tf.constant_initializer(0.0))
            conv1_MS=tf.nn.conv2d(conv2_MS_x4, weights, strides=[1,1,1,1], padding='SAME') + bias
            conv1_MS = lrelu(conv1_MS) 
######################################################### 
#########################################################        
        with tf.variable_scope('layer1_PAN'):
            weights=tf.get_variable("w1_PAN",[3,3,1,16],initializer=tf.truncated_normal_initializer(stddev=1e-3))
            bias=tf.get_variable("b1_PAN",[16],initializer=tf.constant_initializer(0.0))
            conv1_PAN=tf.nn.conv2d(img_PAN, weights, strides=[1,1,1,1], padding='SAME') + bias
            conv1_PAN = lrelu(conv1_PAN) 

######################################################### 
#########################################################    
        MS_dense_2=tf.concat([conv2_MS_x4,conv1_MS,conv1_PAN],axis=-1)
        with tf.variable_scope('layer2_MS'):
            weights=tf.get_variable("w2_MS",[3,3,36,16],initializer=tf.truncated_normal_initializer(stddev=1e-3))
            bias=tf.get_variable("b2_MS",[16],initializer=tf.constant_initializer(0.0))
            conv2_MS=tf.nn.conv2d(MS_dense_2, weights, strides=[1,1,1,1], padding='SAME') + bias
            conv2_MS = lrelu(conv2_MS)  

Here, what's the meaning of [3,3,36,16]?
If I am training with dataset of size 32x32(for LRMS) and 128x128(for PAN), what is this value?
Thank you in advance.

Kind regards.

**Comment by HaoZhang1018** [comment]
> [3,3,36,16]: kernal size=3x3, the number of input channels=36, the number of output channels=16. Theoretically, this setting will not affect the running, even if the patchsize changes. From the code you post, I guess the problem lies in：
MS_x2=tf.image.resize_images(images=img_MS, size=[2self.image_size_MS, 2self.image_size_MS],method=tf.image.ResizeMethod.BICUBIC). Maybe, you can try to modify it as: MS_x2=tf.image.resize_images(images=img_MS, size=[2*32, 2*32],method=tf.image.ResizeMethod.BICUBIC). 
Another option is to pass patchsize as a parameter：
def fusion_model(self,img_MS,img_PAN, MS_size):
    with tf.variable_scope('fusion_model'):
        MS_x2=tf.image.resize_images(images=img_MS, size=[2*MS_size, 2*MS_size],method=tf.image.ResizeMethod.BICUBIC).
The same is true for MS_x4. 
I hope it is helpful to you.

**Comment by KSJhon** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        Thank for your kind reply.
        I have modified your code for WV3 with 8 channels: except adding 4 other placeholders, I changed the ``transfer_model" function as following:
    
        
            
        
    It works well, but I am not sure how well the estimation result is.Thanks again.Good luck.
    
     
    
        
                    ***@***.***
        
    
 

    On 3/7/2022 21:30，Hao ***@***.***> wrote： 






[3,3,36,16]: kernal size=3x3, the number of input channels=36, the number of output channels=16. Theoretically, this setting will not affect the running, even if the patchsize changes. From the code you post, I guess the problem lies in：

MS_x2=tf.image.resize_images(images=img_MS, size=[2self.image_size_MS, 2self.image_size_MS],method=tf.image.ResizeMethod.BICUBIC). Maybe, you can try to modify it as: MS_x2=tf.image.resize_images(images=img_MS, size=[232, 232],method=tf.image.ResizeMethod.BICUBIC).

Another option is to pass patchsize as a parameter：

def fusion_model(self,img_MS,img_PAN, MS_size):

with tf.variable_scope('fusion_model'):

MS_x2=tf.image.resize_images(images=img_MS, size=[2MS_size, 2MS_size],method=tf.image.ResizeMethod.BICUBIC).

The same is true for MS_x4.

I hope it is helpful to you.
